security/fma-shared-lockbox.md

```diff
@@ -0,0 +1,147 @@
+- **Detection:** Verify proxy and implementation contracts
+- **Recovery Path(s):** Pause the system through `SuperchainConfig` to halt the `SharedLockbox`.
+
+### FM6: Unauthorized use of `DEPOSITOR_ACCOUNT` to call `addDepedency` in `DependencyManager`
+
+- **Description:** On L2, only the `DEPOSITOR_ACCOUNT` can call `addDependency` in `DependencyManager`. If attackers can spoof or impersonate that account, they can push arbitrary additions to the dependency set, eventually letting malicious portals into the `SharedLockbox`.
+- **Risk Assessment:** Medium.
+    - Potential Impact: High. Attackers could create fake or malicious chain entries that, if get accepted on L1, would authorize an attacker-controlled portal in `SharedLockbox`.
+    - Likelihood: Low. The `DEPOSITOR_ACCOUNT` is typically used by the derivation pipeline or otherwise restricted. Previously, there were specific deposit calls that used this account, but those are no longer in use. If the `DEPOSITOR_ACCOUNT` becomes callable in other ways under future upgrades, the attack surface would increase.
+- **Mitigations:** There should be tests that ensure `DEPOSITOR_ACCOUNT` only can be called as part of a coordinated upgrade. Review `DEPOSITOR_ACCOUNT` privileges if it is used for other features in the future. Ensure interop fault proof is properly set to reject states that add a non-expected chain.
+- **Detection:** Monitor `DependencyAdded` events to catch unexpected additions. Do periodical checks to `dependencySet` to review the existing dependency set recorded in the chain.
+- **Recovery Path(s):** Because removing a chain from the dependency set is currently irreversible, a hard fork will be the only way to revert an unauthorized addition. Patch any potential issue around `DEPOSITOR_ACCOUNT`.
+
+### FM7: Unable to add new Chains by Reaching the `uint8` limit in `addDependency`
+
+- **Description:** Both `DependencyManager` and `SuperchainConfigInterop` limit the number of dependencies to `type(uint8).max` (= 255). Once the set size reaches 255, any further `addDependency` call will revert.
+- **Risk Assessment:** Low.
+    - Potential Impact: Medium. Future expansions of the interoperable set are blocked.
+    - Likelihood: Low. 255 chains are already quite large, but could be a problem if the interoperable set wants to grow beyond that.
+- **Mitigations:** There should be a defined path to upgrade to increase the type from `uint8` to `uint16` or another approach. Current dependency set size should be monitored.
+- **Detection:** A simple periodic check of `dependencySetSize` in both `DependencyManager` and `SuperchainConfigInterop` should be sufficient to ensure the limit is not approaching. Reverts on calls to `addDependency` would indicate this issue.
+- **Recovery Path(s):** An upgrade over `SuperchainConfigInterop` and `DependencyManager` is needed to support a larger capacity.
+
+### FM8: `ETHMigrated` flag remains `false` in the `OptimismPortalInterop` after migration
```

Comment:
Is the migration plan fleshed out more? Migration is a high-risk action and it would be good to see a concrete sequence of steps we plan to take to perform and validate ETH migration for all portals.

For example one way to mitigate issues here is to atomically set portal implementations to a migrator contract, and then once migration is complete, set it to a portal implementation with no migration features. This removes the need to carry around all migration code and `if (!s.migrated)` checks, and removes the need to remember to remove those in the future

(This may have all been discussed in the design doc phase, in which case feel free to link to those docs. I feel like we discussed this at some point in the past but don't recall where we left off)